batch/test/test_batch.py

```diff
@@ -258,19 +258,19 @@ def test_unsubmitted_state(client: BatchClient):
         j.wait()
 
     builder.submit()
-    with pytest.raises(ValueError):
-        builder.create_job(DOCKER_ROOT_IMAGE, ['echo', 'test'])
+    builder.create_job(DOCKER_ROOT_IMAGE, ['echo', 'test'])
+    assert j.id == 1
```

Comment:
This test needs a new/better name. I might also ask, what is this test really testing now? It seems a bit scattered.